### DISCUSSION TITLE
Advantages of Vue.JS in Themes

### DISCUSSION THREAD

**Discussion by lalzart** [uicrooks/shopify-theme-lab#40] [General]
Hey, I love what you've done here! I have been looking for something like this for a while. This very complete and well documented! Thank you!

Have you found any inherent advantages of using Vue over React for Shopify themes? Or just a personal preference?

**Comment by sergejcodes** [comment]
Hey again 👋

First, yes it's a personal preference.

While you can adjust the configs of this project to handle any JS framework, not just React. I found that it's easiest to work with `Vue` when working on top of an additional templating language, in this case, `liquid`. You can wrap the entire code in a `Vue` instance and basically use all the features `Vue` provides. E.g. passing props is quite easy. Renderless Vue components are also really useful.

**Reply by lalzart** [reply]
Awesome, thank you for everything!

**Reply by haroldao** [reply]
✅

**Reply by sky-xo** [reply]
@sergejcodes out of curiosity... have you tried AlpineJS? I'm using it for my most recent Shopify build and I find it works beautifully with tailwindcss.

**Reply by sergejcodes** [reply]
@jonlemmon just tried it out. Easy to set up and might be just right for smaller projects or projects which focus less on JavaScript overall. 

If we talk about a project that starts out small but then grows in size substantially, I assume there will be a couple of problems:
- There is no big ecosystem for plugins since alpine is quite new
- Shopify doesn't allow any subdirectories in the snippets directory, so you might end up with 100+ files only groupable with a prefix since you will probably use the `.liquid` files as components. 

`Shopify Theme Lab` was designed to handle big and complex projects. Alpine can be a good alternative if you're going for something lighter.
